clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultEventHandler.java

```diff
@@ -129,6 +132,14 @@ public boolean add(final ApplicationEvent event) {
         return applicationEventQueue.add(event);
     }
 
+    @Override
```

Comment:
Was the `addAndGet` method addition here taken from a different commit?

Reply:
Yes, to avoid a cyclic dependency between our changes (integration branch nightmare). This `addAndGet` was included in the PR for [KAFKA-15188](https://issues.apache.org/jira/browse/KAFKA-15188) with more implementation for the PrototypeAsyncConsumer API, that you noted depends on this one. 

Should be the cleanest way to unblock, without having to manually revert changes here as an intermediate step. Let me know if there are any concerns.